### PR TITLE
Update wolfSSL product versions, clone from git tag

### PR DESCRIPTION
**Pull request by cconlon**
This PR updates Yocto/OE recipes to our most current product releases:
- wolfSSL 5.0.0
- wolfMQTT 1.10.0
- wolfTPM 2.3.1
- wolfSSH 1.4.8

As part of the wolfSSL 5.0.0 update, it also updates the `wolfcrypttest` and `wolfcryptbenchmark` example apps using the current versions of `wolfcrypt/test/test.c/h` and `wolfcrypt/benchmark/benchmark.c/h`

This PR also switches the SRC_URI used in recipes to clone from GitHub repo on specific tag, instead of downloading .zip file.

### CODE REVIEW

File: recipes-wolfssl/wolftpm/files/0001-apply-git-PR-196.patch [hunk 1]
```diff
@@ -0,0 +1,370 @@
+diff --git a/examples/pcr/include.am b/examples/pcr/include.am
```
Comment (dgarske) [inline recipes-wolfssl/wolftpm/files/0001-apply-git-PR-196.patch:1]: There is a wolfTPM v2.3.1 tag that can avoid this patch. Would it be easier to just use that? https://github.com/wolfSSL/wolfTPM/releases/tag/v2.3.1

**Review by dgarske — CHANGES_REQUESTED** [review]

File: recipes-wolfssl/wolftpm/files/0001-apply-git-PR-196.patch (hunk 1 above)
Comment (cconlon) [reply recipes-wolfssl/wolftpm/files/0001-apply-git-PR-196.patch:1]: The recipes are currently set up to just download the release .zip's.  We could rework to clone via GitHub, but would probably want to rework for all recipes to be consistent.  If we want to go that route, I can work on adjusting.

Comment (cconlon) [reply recipes-wolfssl/wolftpm/files/0001-apply-git-PR-196.patch:1]: Actually, disregard, maybe we can just switch to downloading the .zip from GitHub, ie:
`https://github.com/wolfSSL/wolfTPM/archive/refs/tags/v2.3.1.zip`.  I'll work to switch this.

Comment (cconlon) [reply recipes-wolfssl/wolftpm/files/0001-apply-git-PR-196.patch]: Recipes have been updated to pull from a git tag, which removes the need for wolfTPM patch file when using tag `v2.3.1`.

**Review by dgarske — APPROVED** [review]

**Pull request merged**